lib/linter.js

```diff
@@ -343,24 +351,23 @@ function normalizeVerifyOptions(providedOptions) {
     return {
         filename: typeof providedFilename === "string" ? providedFilename : "<input>",
         allowInlineConfig: !isObjectOptions || providedOptions.allowInlineConfig !== false,
-        reportUnusedDisableDirectives: isObjectOptions && !!providedOptions.reportUnusedDisableDirectives
+        reportUnusedDisableDirectives: isObjectOptions && !!providedOptions.reportUnusedDisableDirectives,
+        disableFixes: Boolean(providedOptions && providedOptions.disableFixes)
     };
 }
 
 /**
  * Combines the provided parserOptions with the options from environments
  * @param {string} parserName The parser name which uses this options.
- * @param {Object} providedOptions The provided 'parserOptions' key in a config
+ * @param {ParserOptions} providedOptions The provided 'parserOptions' key in a config
  * @param {Environment[]} enabledEnvironments The environments enabled in configuration and with inline comments
- * @returns {Object} Resulting parser options after merge
+ * @returns {ParserOptions} Resulting parser options after merge
  */
 function resolveParserOptions(parserName, providedOptions, enabledEnvironments) {
     const parserOptionsFromEnv = enabledEnvironments
         .filter(env => env.parserOptions)
-        .reduce((parserOptions, env) => ConfigOps.merge(parserOptions, env.parserOptions), {});
-
-    const mergedParserOptions = ConfigOps.merge(parserOptionsFromEnv, providedOptions || {});
-
+        .reduce((parserOptions, env) => lodash.merge(parserOptions, lodash.cloneDeep(env.parserOptions)), {});
```

Comment:
Is it necessary to clone `env.parserOptions` here? I thought `lodash.merge` doesn't mutate the second argument.

Reply:
I wanted to prevent `lodash.merge` modifies `env.parserOptions.nestedProperty`. But, indeed this `lodash.cloneDeep` was unnecesary.